Source/controls/plrctrls.cpp

```diff
@@ -122,61 +123,71 @@ int GetDistanceRanged(Point destination)
 	return Players[MyPlayerId].position.future.ExactDistance(destination);
 }
 
-/**
- * @todo The loops in this function are iterating through tiles offset from the player position. This
- * could be accomplished by looping over the values in the Direction enum and making use of
- * Point math instead of nested loops from [-1, 1].
- */
 void FindItemOrObject()
 {
-	int mx = Players[MyPlayerId].position.future.x;
-	int my = Players[MyPlayerId].position.future.y;
+	Point futurePosition = Players[MyPlayerId].position.future;
 	int rotations = 5;
 
-	// As the player can not stand on the edge of the map this is safe from OOB
-	for (int xx = -1; xx < 2; xx++) {
-		for (int yy = -1; yy < 2; yy++) {
-			if (dItem[mx + xx][my + yy] <= 0)
-				continue;
-			int i = dItem[mx + xx][my + yy] - 1;
-			auto &item = Items[i];
-			if (item.isEmpty()
-			    || item._iSelFlag == 0)
-				continue;
-			int newRotations = GetRotaryDistance({ mx + xx, my + yy });
-			if (rotations < newRotations)
-				continue;
-			if (xx != 0 && yy != 0 && GetDistance({ mx + xx, my + yy }, 1) == 0)
-				continue;
-			rotations = newRotations;
-			pcursitem = i;
-			cursPosition = Point { mx, my } + Displacement { xx, yy };
+	auto searchArea = PointsInRectangleRangeColMajor { Rectangle { futurePosition, 1 } };
+
+	for (Point targetPosition : searchArea) {
+		// As the player can not stand on the edge of the map this is safe from OOB
+		int8_t itemId = dItem[targetPosition.x][targetPosition.y] - 1;
+		if (itemId < 0) {
+			// there shouldn't be any multi-tile objects, but just in case only considering positive indexes here

```

Comment:
Isn't sarcophagus such an object?

Reply:
This comment should say "...multi-tile (dropped) items...". No idea why I wrote object here.